sdk/main/src/Transaction.cc

```diff
@@ -0,0 +1,65 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "Transaction.h"
+
+namespace Hedera
+{
+//-----
+template<typename SdkRequestType>
+const std::chrono::duration<double> Transaction<SdkRequestType>::DEFAULT_VALID_TRANSACTION_DURATION =
+  std::chrono::minutes(2);
+
+//-----
+template<typename SdkRequestType>
+const Hbar Transaction<SdkRequestType>::DEFAULT_MAX_TRANSACTION_FEE = Hbar(2LL);
+
+//-----
+template<typename SdkRequestType>
+SdkRequestType& Transaction<SdkRequestType>::setValidTransactionDuration(const std::chrono::duration<double>& duration)
```

Comment:
Don't you risk objects going out of scope by passing const ref into the setters in this file?

Reply:
The data gets copied into the private member variable and is kept internally, so it wouldn't go out of scope.